src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -916,10 +916,10 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 	 * Add a connection to the active connections list.
 	 */
 	private tryAddActiveConnection(connectionManagementInfo: ConnectionManagementInfo, newConnection: interfaces.IConnectionProfile, addToMru: boolean): void {
-		if (newConnection && addToMru) {
+		if (newConnection) {
 			this._connectionStore.addRecentConnection(newConnection)
 				.then(() => {
-					connectionManagementInfo.connectHandler(true);
+					connectionManagementInfo.connectHandler(addToMru);
```

Comment:
The parameter here is the result of the connection - why does passing addToMru here make sense?

And if addToMru is false then I would expect that we don't add to the recent connections list (recent connections == MRU). So removing the check here doesn't seem like the correct fix.